internal/controller/instaslice_controller.go

```diff
@@ -678,15 +709,22 @@ func (l *RightToLeftPolicy) SetAllocationDetails(profileName string, newStart, s
 	return &inferencev1alpha1.AllocationRequest{}
 }
 
-func (r *InstasliceReconciler) removeInstasliceAllocation(ctx context.Context, instasliceName string, allocation *inferencev1alpha1.AllocationResult) error {
-	if allocation.AllocationStatus.AllocationStatusDaemonset == inferencev1alpha1.AllocationStatusDeleted {
+func (r *InstasliceReconciler) removeInstasliceAllocation(ctx context.Context, instasliceName string, allocResult *inferencev1alpha1.AllocationResult, allocRequest *inferencev1alpha1.AllocationRequest) error {
+	log := logr.FromContext(ctx)
+	if allocResult.AllocationStatus.AllocationStatusDaemonset == inferencev1alpha1.AllocationStatusDeleted {
 		err := utils.UpdateOrDeleteInstasliceAllocations(ctx, r.Client, instasliceName, nil, nil)
 		if err != nil {
 			return err
 		}
 	}
+	// update DeployedPodTotal Metrics by setting value to 0 as pod allocation is deleted
+	if err := r.UpdateDeployedPodTotalMetrics(string(allocResult.Nodename), allocResult.GPUUUID, allocRequest.PodRef.Namespace, allocRequest.PodRef.Name, allocRequest.Profile, 0); err != nil {
```

Comment:
This call should not be in removeInstasliceAllocation method

Reply:
I am worried about these 2 issues if I call it outside:
- reconcile timing issue, removeInstasliceAllocation executes as part of the cleanup, if I call UpdateDeployedPodTotalMetrics after, it may miss it.
- pod lookup issue, if removeInstasliceAllocation removes the allocation, and then later I call UpdateDeployedPodTotalMetrics, the allocation may no longer exist in the Instaslice object.